tests/system/Database/Live/WriteTypeQueryTest.php

```diff
@@ -28,14 +28,33 @@ final class WriteTypeQueryTest extends CIUnitTestCase
     protected $refresh = true;
     protected $seed    = CITestSeeder::class;
 
+    /**
+     * Whether CodeIgniter ignores RETURNING for isWriteType.
+     *
+     * Currently, only Postgre is supported by CodeIgniter.
+     * This method should be updated if support for RETURNING
+     * is expanded to other databases.
+     *
+     * @param string $dbDriver
+     */
+    private function testAssertTypeReturning($dbDriver): string
+    {
+        $assertType = 'assertTrue';
+        if ($dbDriver === 'Postgre') {
+            $assertType = 'assertFalse';
+        }
+
+        return $assertType;
+    }
```

Comment:
Test codes with changing assertions are difficult to understand. 
Because we cannot understand the code `$this->{$assertType}($this->db->isWriteType($sql))` by just looking at it. Please remove such code.

Also, it is pointless to test for RETURNING, since all but Postgres do not support RETURNING.

Please change this method to skip testing except for Postgres.

Reply:
Hello,

Instead of a dynamic $assertType, I have now wrapped an if statement around the assertFalse and assertTrue calls when testReturning is needed. Because each test is now its own function, had I only tested returning statements for Postgre, it would have returned those 15 tests as Risky (I don't know if this means it would have failed).

While it is true that CodeIgniter currently does not support RETURNING in other databases, it is not true that other databases that CodeIgniter otherwise supports do not support RETURNING:

https://www.sqlite.org/lang_returning.html
https://mariadb.com/kb/en/insertreturning/

(It looks like MySQL does not, which makes supporting MariaDB's RETURNING statements complicated.)

I could change this pull request to just always support RETURNING as not isWriteType, and not have a separate override for Postgre. Ultimately, whether isWriteType returns TRUE or FALSE in database systems that don't support it is probably irrelevant, as the query should return an error anyway.

Reply:
> I could change this pull request to just always support RETURNING as not isWriteType, and not have a separate override for Postgre. Ultimately, whether isWriteType returns TRUE or FALSE in database systems that don't support it is probably irrelevant, as the query should return an error anyway.

Yes, you are correct!
If there is a `RETURNING` clause in a statement, it should be a NOT write type query.
Then `testReturning()` would be unnecessary.

Reply:
Ok, this pull request has been modified accordingly.

Reply:
Thank you!